### PR TITLE
[gha] fix up rustdoc for consistence & correct trigger.  Fix up lint to not rerun on bors land.

### PR DESCRIPTION
**Pull request by rexhoffman**
## Motivation

Clean up straggling issues in the ci/cd pipeline.   get rust docs running again.

### Have you read the [Contributing Guidelines on pull requests]

Yes

## Test Plan

CI

## Related PRs

None

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

File: .github/workflows/ci-post-land.yml [hunk 1]
```diff
@@ -56,14 +56,14 @@ jobs:
...
     needs: prepare
+    if: ${{ needs.prepare.outputs.rust-changes == 'true' }}
     environment:
       name: Sccache
     runs-on: ubuntu-latest-xl
     container:
       image: diem/build_environment:${{ needs.prepare.outputs.changes-target-branch }}
       volumes:
         - "${{github.workspace}}:/opt/git/diem"
-    if: ${{ needs.prepare.outputs.rust-changes == 'true' }}
```
Comment (rexhoffman) [inline .github/workflows/ci-post-land.yml:66]: just moving the "if" line up to be consistent with other jobs.

File: .github/workflows/ci-post-land.yml [hunk 2]
```diff
@@ -329,31 +329,31 @@ jobs:
 
   rustdoc:
     needs: prepare
-    if: ${{ needs.prepare.output.changes-target-branch == 'main' }}
-    env:
-      CARGO_INCREMENTAL: 0
-      RUSTFLAGS: -D warnings
```
Comment (rexhoffman) [inline .github/workflows/ci-post-land.yml:335]: Moving down to the cargo line for cleanliness.

File: .github/workflows/ci-post-land.yml [hunk 3]
```diff
@@ -329,31 +329,31 @@ jobs:
...
+    if: ${{ needs.prepare.outputs.changes-target-branch == 'main' && needs.prepare.outputs.rust-changes == 'true'}}
```
Comment (rexhoffman) [inline .github/workflows/ci-post-land.yml:332]: Triggering on the right input "output" -> "outputs", kept this from firing before.    Also, don't rerun if no rust changes were found.

File: .github/workflows/ci-post-land.yml [hunk 4]
```diff
@@ -329,31 +329,31 @@ jobs:
...
         with:
-          toolchain: stable
-          override: true
-
-      # Build the rust crate docs
-      # Use `RUSTC_BOOTSTRAP` in order to use the `--enable-index-page` flag of rustdoc
-      # This is needed in order to generate a landing page `index.html` for workspaces
+          path: "/opt/cargo/git\n/opt/cargo/registry\n/opt/cargo/.package-cache"
+          key: crates-${{ runner.os }}-${{ hashFiles('Cargo.lock') }}
+          restore-keys: "crates-${{ runner.os }}"
```
Comment (rexhoffman) [inline .github/workflows/ci-post-land.yml:346]: Make consistent with other rust job's runtime (use the docker image and prepare, and cache step).

File: .github/workflows/ci-post-land.yml [hunk 5]
```diff
@@ -329,31 +329,31 @@ jobs:
...
       - name: Build Documentation
+        # Build the rust crate docs
+        # Use `RUSTC_BOOTSTRAP` in order to use the `--enable-index-page` flag of rustdoc
+        # This is needed in order to generate a landing page `index.html` for workspaces
```
Comment (rexhoffman) [inline .github/workflows/ci-post-land.yml:350]: Move the doc in to the yaml block.

File: .github/workflows/ci-post-land.yml [hunk 6]
```diff
@@ -329,31 +329,31 @@ jobs:
...
         run: cargo doc --no-deps --workspace --lib
         env:
           RUSTC_BOOTSTRAP: 1
           RUSTDOCFLAGS: "-Z unstable-options --enable-index-page"
-
+          CARGO_INCREMENTAL: 0
+          RUSTFLAGS: -D warnings
```
Comment (rexhoffman) [inline .github/workflows/ci-post-land.yml:356]: Flags from above moved in to only cargo step.

File: .github/workflows/ci-test.yml [hunk 7]
```diff
@@ -94,15 +95,14 @@ jobs:
...
-          output=None
-          if ${{ github.event_name == 'push' }}; then
-            echo "will use sccache environment for rust builds";
-            output=Sccache
+          res=true
+          if [[ ! -f "$CHANGES_CHANGED_FILE_OUTPUTFILE" ]] || [[ "$(cat "$CHANGES_CHANGED_FILE_OUTPUTFILE" | wc -l)" == 0 ]]; then
+            res=false;
           fi
-          echo "::set-output name=name::$(echo $output)";
+          echo "::set-output name=changes-found::$(echo $res)";
```
Comment (rexhoffman) [inline .github/workflows/ci-test.yml:105]: Environment was never used, and I should have deleted it a while back.   Now there is a test to see if any files changed in a pull request.   If any changed, output true.   Bors rebase on land will result in prs firing with no changes.

File: .github/workflows/ci-test.yml [hunk 8]
```diff
@@ -497,7 +497,7 @@ jobs:
     runs-on: ubuntu-latest-xl
     timeout-minutes: 30
     needs: prepare
-    #if: ${{ needs.prepare.outputs.test-rust == 'true' }}
+    if: ${{ needs.prepare.outputs.any-changes-founds == 'true' }}
```
Comment (rexhoffman) [inline .github/workflows/ci-test.yml:500]: Don't run lint step on a bors rebase.

**Review by bmwill — APPROVED** [review]

**Pull request merged**